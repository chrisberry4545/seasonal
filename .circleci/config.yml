version: 2.1

commands:
  install-dependencies:
    steps:
      - restore_cache:
          name: Restore dependency cache
          keys:
            - seasonal-deps-{{ checksum "package-lock.json" }}
      - run:
          name: Install dependencies
          command: |
            npm i
            npm run bootstrap
      - save_cache:
          name: Save dependency cache
          paths:
            - node_modules
          key: seasonal-deps-{{ checksum "package-lock.json" }}

jobs:
  backend_test:
    docker:
      - image: circleci/node:10.15
    steps:
      - checkout
      - setup_remote_docker
      - run:
          name: Run backend tests
          command: ./tests/backend-run-functional-tests.sh

  backend_deploy:
    docker:
      - image: circleci/node:10.15
    steps:
      - checkout
      - install-dependencies
      - run:
          name: Copy env config
          command: |
            cd ./packages/backend
            cp .env-ci.yml .env.yml
      - run:
          name: Build dist version
          command: |
            cd ./packages/backend
            npm run build
            npm run move-dist-to-top-level
      - run:
          name: Install Serverless CLI
          command: sudo npm i -g serverless
      - run:
          name: Deploy backend
          command: |
            cd ./packages/backend
            sls deploy --stage prod

  frontend_deploy:
    docker:
      - image: circleci/node:10.15
    steps:
      - checkout
      - install-dependencies
      - add_ssh_keys:
          fingerprints:
            - cd:42:e8:de:36:e3:c0:f9:0d:6f:55:27:47:57:ab:4b
      - run:
          name: Git configuration
          command: |
            git config user.email "chrisbery4545@googlemail.com"
            git config user.name "chrisbery4545"
      - run:
          name: Deploy frontend
          command: |
            cd ./packages/frontend
            npm run deploy

workflows:
  version: 2
  backend:
    jobs:
      - backend_test
      - backend_approval:
          type: approval
          filters:
            branches:
              only:
                - master
          requires:
           - backend_test
      - backend_deploy:
          requires:
            - backend_approval
      - frontend_approval:
          type: approval
          filters:
            branches:
              only:
                - master
      - frontend_deploy:
          requires:
            - frontend_approval
