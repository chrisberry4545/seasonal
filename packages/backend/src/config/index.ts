export * from './env-vars';

export * from './urls';
