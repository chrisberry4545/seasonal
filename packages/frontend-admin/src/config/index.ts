const BACKEND_URL = process.env.BACKEND_URL
  || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/admin';

export const LOGIN_URL =
  process.env.LOGIN_URL || `${BACKEND_URL}/login`;

export const RECIPES_URL =
  process.env.RECIPES_URL || `${BACKEND_URL}/recipes`;
