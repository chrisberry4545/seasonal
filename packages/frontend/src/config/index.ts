const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
    || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/v1';

export const SEASON_DATA_URL =
  process.env.REACT_APP_SEASON_DATA_URL || `${BACKEND_URL}/season-data`;

export const FOOD_DETAILS_DATA_URL =
  process.env.REACT_APP_FOOD_DETAILS_DATA_URL || `${BACKEND_URL}/food-data`;
