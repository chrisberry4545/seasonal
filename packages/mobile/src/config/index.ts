import { Platform } from 'react-native';

export const PRIVACY_POLICY_URL: string =
  'https://chrisberry4545.github.io/eat-seasonal-privacy-policy/';
export const STORE_URL: string = Platform.OS === 'ios'
  ? 'https://play.google.com/store/apps/details?id=com.chrisbdev.seasonal'
  : 'market://details?id=com.chrisbdev.seasonal';
