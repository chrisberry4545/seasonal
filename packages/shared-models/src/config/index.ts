const BACKEND_URL = process.env.REACT_APP_BACKEND_URL
  || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/v2';

export const SEASON_URL =
  process.env.SEASON_URL || `${BACKEND_URL}/season`;

export const FOOD_DETAILS_URL =
  process.env.FOOD_DETAILS_URL || `${BACKEND_URL}/food`;

export const SEASON_WITH_FOOD_URL =
  process.env.SEASON_WITH_FOOD_URL || `${BACKEND_URL}/season-with-food`;

export const SEASON_WITH_RECIPES_URL =
  process.env.SEASON_WITH_RECIPES_URL || `${BACKEND_URL}/season-with-recipes`;

export const COUNTRY_URL =
  process.env.COUNTRY_URL || `${BACKEND_URL}/country`;
