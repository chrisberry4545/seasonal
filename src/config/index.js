const BACKEND_URL = process.env.BACKEND_URL
    || 'http://localhost:3000';

const config = {
    SEASON_DATA_URL: process.env.SEASON_DATA_URL || `${BACKEND_URL}/season-data`
};

export {
    config
};
