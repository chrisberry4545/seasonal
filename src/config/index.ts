const BACKEND_URL: string = process.env.BACKEND_URL
  || 'https://6usraevcmc.execute-api.eu-west-2.amazonaws.com/prod/v1';

export const SEASON_DATA_URL: string =
  process.env.SEASON_DATA_URL || `${BACKEND_URL}/season-data`;

export const FOOD_DETAILS_DATA_URL =
  process.env.FOOD_DETAILS_DATA_URL || `${BACKEND_URL}/food-data`;

export const PRIVACY_POLICY_URL: string =
  'https://chrisberry4545.github.io/eat-seasonal-privacy-policy/';
