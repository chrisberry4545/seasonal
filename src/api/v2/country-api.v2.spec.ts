import {
  app
} from '../../app';

import supertest, { Response } from 'supertest';
import { V2_ENDPOINT, COUNTRY_ENDPOINT } from '../../config';
import { ICountry, IRegion } from '@chrisb-dev/seasonal-shared';

const countryUrl = `${V2_ENDPOINT}/${COUNTRY_ENDPOINT}`;
const getAllRegionsFromCountries = (countries: ICountry[]): IRegion[] => (
  countries.reduce((regions, country) => [
    ...regions,
    ...country.regions
  ], [] as IRegion[])
);

describe('Get countries', () => {
  let response: Response;
  beforeAll(async () => {
    response = await supertest(app).get(`/${countryUrl}`);
  });

  test('Returns a status of 200', () => {
    expect(response.status).toBe(200);
  });
  test('Retrieves a single food item', () => {
    expect(response.body).toMatchSnapshot();
  });
  test('Returns multiple countries', () => {
    expect(response.body.length > 0).toBe(true);
  });
  test('Populates every countries id', () => {
    const areAllIdsDefined =
      response.body.every((country: ICountry) => country.id !== undefined);
    expect(areAllIdsDefined).toBe(true);
  });
  test('Populates every countries name', () => {
    const areAllNamesDefined =
      response.body.every((country: ICountry) => country.name !== undefined);
    expect(areAllNamesDefined).toBe(true);
  });
  test('Every countries has regions', () => {
    const doAllCountriesHaveRegions =
      response.body.every((country: ICountry) => country.regions.length > 0);
    expect(doAllCountriesHaveRegions).toBe(true);
  });
  test('Populates every regions name', () => {
    const allRegionsNamesDefined = getAllRegionsFromCountries(response.body)
      .every((region) => region.name !== undefined);
    expect(allRegionsNamesDefined).toBe(true);
  });
  test('Populates every regions code', () => {
    const allRegionsCodesDefined = getAllRegionsFromCountries(response.body)
      .every((region) => region.code !== undefined);
    expect(allRegionsCodesDefined).toBe(true);
  });
});
