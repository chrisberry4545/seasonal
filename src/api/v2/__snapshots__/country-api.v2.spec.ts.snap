// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get countries Retrieves a single food item 1`] = `
Array [
  Object {
    "id": "GBR",
    "name": "Great Britain & Ireland",
    "regions": Array [
      Object {
        "code": "gbr",
        "latLng": Object {
          "lat": 51.5074,
          "lng": -0.118092,
        },
        "name": "Great Britain & Ireland",
      },
    ],
  },
  Object {
    "id": "AUS",
    "name": "Australia",
    "regions": Array [
      Object {
        "code": "aus-melbourne",
        "latLng": Object {
          "lat": -37.840935,
          "lng": 144.946457,
        },
        "name": "Melbourne",
      },
      Object {
        "code": "aus-perth",
        "latLng": Object {
          "lat": -31.953512,
          "lng": 115.857048,
        },
        "name": "Perth",
      },
      Object {
        "code": "aus-sydney",
        "latLng": Object {
          "lat": -33.865143,
          "lng": 151.2099,
        },
        "name": "Sydney",
      },
    ],
  },
]
`;
